packages/features/embed/Embed.tsx

```diff
@@ -1020,40 +1020,54 @@ const EmbedTypeCodeAndPreviewDialogContent = ({
             </div>
           )}
         </div>
-        <div className="flex w-2/3 flex-col px-8 pt-8">
+        <div className="flow-y-scroll flex h-[95vh] w-2/3 flex-col px-8 pt-8">
           <HorizontalTabs
             data-testid="embed-tabs"
-            tabs={embedType === "email" ? parsedTabs.filter((tab) => tab.name === "Preview") : parsedTabs}
+            tabs={parsedTabs.filter((tab) => tab.name !== "Preview")}
             linkShallow
           />
           {tabs.map((tab) => {
+            const previewTab = tabs.find((tab) => tab.name === "Preview");
             if (embedType !== "email") {
               return (
                 <div
                   key={tab.href}
                   className={classNames(
                     searchParams?.get("embedTabName") === tab.href.split("=")[1]
-                      ? "flex flex-grow flex-col"
+                      ? "flex h-full flex-col"
                       : "hidden"
                   )}>
-                  <div className="flex h-[55vh] flex-grow flex-col">
-                    {tab.type === "code" ? (
-                      <tab.Component
-                        namespace={namespace}
-                        embedType={embedType}
-                        calLink={calLink}
-                        previewState={previewState}
-                        ref={refOfEmbedCodesRefs.current[tab.name]}
-                      />
-                    ) : (
-                      <tab.Component
-                        namespace={namespace}
-                        embedType={embedType}
-                        calLink={calLink}
-                        previewState={previewState}
-                        ref={iframeRef}
-                      />
-                    )}
+                  <div className="flex h-full flex-col">
+                    <div className="flex flex-1 flex-col">
+                      {tab.type === "code" ? (
+                        <tab.Component
+                          namespace={namespace}
+                          embedType={embedType}
+                          calLink={calLink}
+                          previewState={previewState}
+                          ref={refOfEmbedCodesRefs.current[tab.name]}
+                        />
+                      ) : (
+                        <tab.Component
+                          namespace={namespace}
+                          embedType={embedType}
+                          calLink={calLink}
+                          previewState={previewState}
+                          ref={iframeRef}
+                        />
+                      )}
+                    </div>
+                    <div className="flex-1">
+                      {previewTab && (
+                        <previewTab.Component
```

Comment:
This is causing a preview tab per other tab. So, 2 preview tabs. We should have only one preview tab as there is a communication that happens with iframe and having 2 different iframes would have different results for different tabs.

Reply:
I didn't realise that, made some changes please review again